### PR TITLE
Feat/homepage

### PR DESCRIPTION
**Pull request by alextio**
- Added basic responsive layout for front page
- Added basic reponsive typography for hero section

localhost_3000_(Macbook pro)

* :sparkles: sort publication list

* :sparkles: support PDF links in  component

* :clown_face: add mock data

* :recycle: refactor state management for filtering with `useRouter`

* :speech_balloon: update filter name for publication type

* :goal_net: fix error

- Write a one-line summary of what you did.
- Add a screenshot/video of the feature you developed (if applicable).

### CODE REVIEW

File: src/data/members.ts [hunk 1]
```diff
@@ -16,6 +19,11 @@ class Member {
   get fullName() {
     return `${this.name} ${this.lastName}`
   }
+
+  get imgPath() {
+    let dirPath = '/images/members/'
+    return `${dirPath}${this.img}`
```
Comment (jhw123) [inline src/data/members.ts:25]:
```suggestion
    return `/images/members/${this.img}`
```

File: src/data/publications.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export const ResearchTopics = [
   'social',
   'human-ai-interaction',
 ] as const
-export type ResearchTopic = (typeof ResearchTopics)[number]
+export type ResearchTopicType = (typeof ResearchTopics)[number]
```
Comment (jhw123) [inline src/data/publications.ts:15]: Why did you add `Type` at the end? Was there a conflicting name?

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,3 +1,403 @@
...
+    z-index: 0;
+  `
+
+  return (
+    <Section id="hero-section" style={{ padding: '0' }}>
+      {/* padding: 0 is to allow image to stretch to the right side of the webpage*/}
+      <HeroContainer>
+        <HeroTextArea id="hero-text-area">
+          <HeroTitle>
+            Welcome to <span style={{ color: `${Color.orange900}`, paddingRight: '7px' }}>Kixlab</span>!
```
Comment (jhw123) [inline src/app/page.tsx:140]: I would just put a whitespace after KIXLAB instead of paddingRight because the padding should be responsive to the font size ideally.

**Review by jhw123 — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (alextio) [reply src/app/page.tsx:140]: I am using the CSS property `text-transform: uppercase` to make the title all uppercase. Is it better to make the text itself uppercase?

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,3 +1,403 @@
...
+  const HeroImageContainer = styled.div`
+    flex-basis: 50%;
+    display: grid;
+    align-content: center;
+    position: relative;
+    z-index: 0;
+  `
+
+  return (
+    <Section id="hero-section" style={{ padding: '0' }}>
```
[the remark this answers is not in the document]
Comment (alextio) [reply src/app/page.tsx:135]: The id is mainly for debugging purposes. It helps to quickly find the right section in the DOM. 

Should I make an entire new styled component for the Hero section then?

File: src/app/page.tsx [hunk 5]
```diff
@@ -1,3 +1,403 @@
...
+        ))}
+      </MediaArea>
+    </Section>
+  )
+}
+
 export default function Page() {
-  return <main></main>
+  return (
+    <main style={{ padding: '0px', gap: '0px', marginBottom: '16px', alignItems: 'center' }}>
```
[the remark this answers is not in the document]
Comment (alextio) [reply src/app/page.tsx:396]: Screenshot 2024-05-02 at 1 21 40 PM
^This is without `padding: '0px'`
Screenshot 2024-05-02 at 1 22 43 PM
^ This is without `gap: '0px'` Notice the small white gap between the news and hero section

File: src/app/page.tsx (hunk 3 above)
Comment (alextio) [reply src/app/page.tsx:140]: [
Screenshot 2024-05-02 at 1 37 05 PM
](url)
If I don't use `paddingRight`, this is what the webpage will look like in certain cases.

File: src/data/publications.ts (hunk 2 above)
Comment (alextio) [reply src/data/publications.ts:15]: Earlier in the file, there was a `export type PublicationType = (typeof PublicationTypes)[number]` line, so I renamed `ResearchTopic` to `ResearchTopicType` in order to maintain consistency. Plus, I think `ResearchTopicType` is a more accurate name.

File: src/app/page.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhw123) [reply src/app/page.tsx:140]: I think so. `text-transform` is often used for variable texts.

Comment (jhw123) [reply src/app/page.tsx:140]: good point. I haven't thought of this case.

File: src/app/page.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (jhw123) [reply src/app/page.tsx:396]: The padding is resulted from the global style set by the browser, right? If so, I think we should add padding: 0 to the reset css.

File: src/app/homepage/Styles.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  grid-template-columns: ${props => props.columnTemplate || '1fr 1fr'}; // default is two columns
+`
+export const FlexContainer = styled.div<{ direction?: 'row' | 'column'; gap?: string }>`
+  display: flex;
+  flex-direction: ${props => props.direction || 'row'};
+  gap: ${props => props.gap || '0px'}
+  align-items: center;
+  max-width: 100vw;
+  justify-content: space-evenly;
+  // flex-wrap: wrap;
```
Comment (jhw123) [inline src/app/homepage/Styles.tsx:22]: remove commented line

Comment (alextio) [reply src/app/homepage/Styles.tsx:22]: https://github.com/kixlab/website/pull/19/commits/6fab729593ba73a681079b9ff5ea19892dc5d19c

**Review by jhw123 — APPROVED** [review]
